app/webpacker/components/ContactEditProfilePage/EditProfileForm.jsx

```diff
@@ -0,0 +1,185 @@
+  const profileDetails = data?.data?.person;
+
+  useEffect(() => {
+    setEditedProfileDetails(profileDetails);
+  }, [profileDetails]);
+
+  const formSubmitHandler = () => {
+    const formData = new FormData();
+    formData.append('formValues', JSON.stringify({
+      editedProfileDetails, editProfileReason, wcaId, contactEmail,
+    }));
+    formData.append('attachment', proof);
+    save(
+      contactEditProfileActionUrl,
+      formData,
+      onContactSuccess,
+      { method: 'POST', headers: {}, body: formData },
+      setSaveError,
+    );
+  };
+
+  const handleEditProfileReaconChange = (e, { value }) => {
+    setEditProfileReason(value);
+  };
+
+  const handleProofUpload = (event) => {
+    setProof(event.target.files[0]);
+  };
+
+  const handleFormChange = (e, { name: formName, value }) => {
+    setEditedProfileDetails((prev) => ({ ...prev, [formName]: value }));
+  };
+
+  if (saving || isLoading) return <Loading />;
+  if (saveError || isError) return <Errored />;
+
+  return (
+    <Form onSubmit={formSubmitHandler}>
+      {!loggedInUserData && (
+        <Form.Input
+          label={i18n.t('page.contact_edit_profile.form.contact_email.label')}
+          value={contactEmail}
+          onChange={(e, { value }) => setContactEmail(value)}
+          required
+        />
+      )}
+      {!loggedInUserWcaId && (
+        <Form.Field
+          control={IdWcaSearch}
+          name="wcaId"
+          label={i18n.t('page.contact_edit_profile.form.wca_id.label')}
+          value={customWcaId}
+          onChange={(e, { value }) => setCustomWcaId(value)}
+          disabled={customWcaId}
+          multiple={false}
+          model={SEARCH_MODELS.person}
+          required
+        />
+      )}
+      {wcaId && (
+        <>
+          <Form.Input
+            label={i18n.t('activerecord.attributes.user.name')}
+            name="name"
+            value={editedProfileDetails?.name}
+            onChange={handleFormChange}
+          />
+          <Form.Select
+            options={countryOptions}
+            label={i18n.t('activerecord.attributes.user.country_iso2')}
+            name="country_iso2"
+            search
+            value={editedProfileDetails?.country_iso2}
+            onChange={handleFormChange}
+          />
+          <Form.Select
+            options={genderOptions}
+            label={i18n.t('activerecord.attributes.user.gender')}
+            name="gender"
+            value={editedProfileDetails?.gender}
+            onChange={handleFormChange}
+          />
+          <Form.Field
+            label={i18n.t('activerecord.attributes.user.dob')}
+            name="dob"
+            control={UtcDatePicker}
+            showYearDropdown
+            dateFormatOverride="YYYY-MM-dd"
+            dropdownMode="select"
+            isoDate={editedProfileDetails?.dob}
+            onChange={(date) => handleFormChange(null, {
+              name: 'dob',
+              value: date,
+            })}
```

Comment:
Can you put this into a separate callback please?
In general, `onChange` and similar React component callbacks should only be written in-line if they can (comfortably) fit on one line

Reply:
Done